src/routing_node.rs

```diff
@@ -190,8 +191,9 @@ impl RoutingNode {
 
     /// When CRUST reports a lost connection, ensure we remove the endpoint anywhere
     fn handle_lost_connection(&mut self, endpoint : Endpoint) {
-        error!("Lost connection on {:?}, but CORE IS NOT UPDATED!", endpoint);
-        //unimplemented!()
+        error!("Lost connection on {:?}", endpoint);
+        let connection_name = self.core.lookup_endpoint(&endpoint); 
```

Comment:
It is possibly more desirable to have an explicit function core::drop_endpoint (currently only have drop_peer), but that's up for discussion